examples/shopping_cart.rs

```diff
@@ -0,0 +1,72 @@
+#![feature(type_alias_enum_variants)]
+
+#[macro_use]
+extern crate mirai_annotations;
+
+pub mod foreign_contracts;
+
+struct Cart {
+    items: Vec<i32>
+}
+
+impl Cart {
+    fn new() -> Cart {
+        Cart {
+            items: Vec::new(),
+        }
+    }
+
+    fn add_item(&mut self, value: i32) {
+        precondition!(value > 0);
```

Comment:
This is appropriate. The type system cannot express this, so the precondition adds value and gives MIRAI something to check at call sites.